app/assets/v2/js/quadraticlands/mission/diplomacy/room.js

```diff
@@ -0,0 +1,248 @@
+    i.addEventListener('input', input => {
+      if (i.value < 0) {
+        i.value = 0;
+      }
+
+      updateVoucheBar();
+
+      console.debug("USED", window.used)
+      console.debug("USE", window.use)
+      console.debug("AWAILABLE", (window.walletbalance - window.used) )
+
+      if (window.use > (window.walletbalance - window.used) )
+      {
+        console.debug("you can not use so much gtc");
+        i.value = 0;
+        updateVoucheBar();
+      }
+
+    });
+  });
+
+});
+
+
+function updateVoucheBar(){
+
+  window.use = 0;
+  inputs = document.querySelectorAll('.member-card .front input');
+
+  inputs.forEach(i => {
+    if (i.value > 0) {
+      window.use += Number(i.value);
+    }
+  });
+
+  diplomacy_wallet_use = document.getElementById('diplomacy_wallet_use');
+  diplomacy_wallet_use.innerHTML = window.use;
+
+  diplomacy_wallet_available = document.getElementById('diplomacy_wallet_available');
+  diplomacy_wallet_available.innerHTML = window.walletbalance - window.used;
+  console.debug("updateVouceBar");
+
+}
+
+// fetch the gtc balance
+async function diplomacyWallet(){
+  console.debug("diplomacy wallet");
+  try {
+
+    let balance = await getTokenBalances(gtc_address());
+
+    diplomacy_wallet_balance.innerHTML = balance.balance.toFixed(2);
+    diplomacy_wallet_address.innerHTML = truncate(selectedAccount);
+    window.walletbalance = balance.balance.toFixed(2);
+    console.debug("walletbalance", window.walletbalance)
+
+    // to calc available balance
+    updateVoucheBar();
+
+  } catch (e) {
+    console.error(e);
+  }
+}
+
+
+// reads all the members input fields and generate a nice
+// array of objects to do a "sign" + a "safe to db"
+// not sure how to do this part.
+
+async function vouche() {
+
+  console.debug('VOUCHE');
+
+  // read all input fields what have data-member
+  members = document.querySelectorAll('[data-member]');
+
+  // define a result array
+  const result = [];
+
+  // push all vouche data of each member to the result
+  members.forEach(member => {
+    var entry = {
+      userid: member.dataset.userid,
+      username: member.dataset.username,
+      value: member.value
+    };
+
+    result.push(entry);
+  });
+
+  // @kev : do something with the result ( sign, safe, whatever)
+  // this is how far i could come. now your turn :)
+  const accounts = await web3.eth.getAccounts();
+  const account = accounts[0];
+  const package = {
+    'votes': result,
+    'balance': balance,
+    'account': account,
+  }
+  let signature = await web3.eth.personal.sign(JSON.stringify(package) , account);
+  const diplomacy_wallet_balance = document.getElementById('wallet_token_balance');
+  const params = {
+    package: package,
+    signature: signature,
+    csrfmiddlewaretoken: $('[name=csrfmiddlewaretoken]').val(),
+  }
+  const url = document.location.href;
+  $.ajax({
+    type: "POST",
+    url: url,
+    data: params,
+    success: function(response){
+      alert("Vote submitted")
```

Comment:
hey @scco how do i make this into a fancy top right message like you do when u presnt the user a message?